listeners/measured.go

```diff
@@ -46,11 +96,23 @@ func (c *wrapMeasuredConn) OnState(s http.ConnState) {
 // Responds to the "measured" message type
 func (c *wrapMeasuredConn) ControlMessage(msgType string, data interface{}) {
 	if msgType == "measured" {
-		c.Conn.ID = data.(string)
+		ctxUpdate := data.(map[string]interface{})
+		c.ctxMx.Lock()
+		newContext := make(map[string]interface{}, len(c.ctx))
+		// Copy context
+		for key, value := range c.ctx {
+			newContext[key] = value
+		}
+		// Update context
+		for key, value := range ctxUpdate {
+			newContext[key] = value
+		}
```

Comment:
The context doesn't store numeric values, it stores stuff like origin, client_ip, etc, so there's no need to perform calculations.

In practice, where we use it, yes we overwrite the entire context everytime. And actually, in practice, we only set this once because the opsfilter is only called once. However, this package is in the public http-proxy project and this could in theory be used in other places, so I figured that merging context keys was the best general behavior.